zendesk/zd_ticket_comments_service.go

```diff
@@ -0,0 +1,106 @@
+package zendesk
+
+import (
+	"fmt"
+	"time"
+)
+
+// Ticket represents a Zendesk Ticket.
+//
+// Zendesk Core API docs: https://developer.zendesk.com/rest_api/docs/core/ticket_comments
+
+type TicketComment struct {
+	ID          int64        `json:"id,omitempty"`
+	Type        string       `json:"type,omitempty"`
+	Body        string       `json:"body,omitempty"`
+	HTMLBody    string       `json:"html_body,omitempty"`
+	PlainBody   string       `json:"plain_body,omitempty"`
+	Public      bool         `json:"public,omitempty"`
+	AuthorID    int64        `json:"author_id,omitempty"`
+	Attachments []Attachment `json:"attachments,omitempty"`
+	Via         *Via         `json:"via,omitempty"`
+	MetaData    interface{}  `json:"metadata,omitempty"`
+	CreatedAt   *time.Time   `json:"created_at,omitempty"`
+	Uploads     []string     `json:"uploads,omitempty"`
+}
+
```

Comment:
we have comments and attachments defined in the **zd_ticket_service.go**, why you create a new one?

Reply:
If you read closely of https://developer.zendesk.com/rest_api/docs/support/tickets
you will see "comments" and "attachments" are not supposed in ticket struct. 
That is the first reason why you see the decoupled ticket_comments_service.go

Another more important reason is to align with the sql table structure, which naturally requires decoupling "comments table" and "ticket table" for performance consideration.